### DISCUSSION TITLE
Jacobian neural network (weights)

### DISCUSSION THREAD

**Discussion by gerdm** [google/flax#1393] [Q&A] [answered]
Is there any way to compute the Jacobian of a neural network with respect to each of the weights using Flax?

I only found [this](https://github.com/google/flax/issues/735) issue as a reference, but the Jacobian is computed in terms of the input values, whereas I would like to compute the Jacobian with respect to each of the weights.

I also tried the following code, but it didn't work.

```python
class MLP1D(nn.Module):
    n_hidden: int
        
    def setup(self):
        self.layer1 = nn.Dense(self.n_hidden)
        self.layer2 = nn.Dense(self.n_hidden)
        self.layer_out = nn.Dense(1)
    
    def __call__(self, x):
        x = x[:, None]
        x = nn.tanh(self.layer1(x))
        x = self.layer_out(x)
        return x
    
model = MLP1D(3)
    
Hw = jax.jacobian(model)
Hw(x)
```

**Comment by jheek** [comment, marked as answer]
With the model in your example the following should work:

```
variables = model.init(rng, inputs)

Hw = jax.jacobian(model.apply)
Hw(variables, inputs)
```
